code/obj/item/gun/kinetic.dm

```diff
@@ -870,6 +880,267 @@ ABSTRACT_TYPE(/obj/item/survival_rifle_barrel)
 			projectiles = list(current_projectile)
 			UpdateIcon()
 
+/obj/item/gun/kinetic/uzi
```

Comment:
Swapping ammo types between standard 9mm and the custom uzi ammo makes the burst mode turn into a single shot, switching to full auto and back fixes it. Don't think this is intended?

Reply:
ill double check this. its to do with ammo and firemodes being the same thing ):